src/stream.cpp

```diff
@@ -997,6 +1004,47 @@ namespace stream {
     // Video traffic is sent on this thread
     platf::adjust_thread_priority(platf::thread_priority_e::high);
 
+    struct {
+      using steady_clock = std::chrono::steady_clock;
+      struct {
+        steady_clock::time_point last_print_time = steady_clock::now();
+        uint16_t latency_min = std::numeric_limits<uint16_t>::max();
+        uint16_t latency_max = 0;
+        uint32_t latency_total = 0;
+        uint32_t calls = 0;
+      } data;
+
+      uint16_t
+      convert_latency(const std::chrono::steady_clock::duration &duration) const {
+        const auto duration_us = std::chrono::duration_cast<std::chrono::microseconds>(duration).count();
+        return (uint16_t) std::clamp<decltype(duration_us)>((duration_us + 50) / 100, 0, std::numeric_limits<uint16_t>::max());
+      }
+
+      void
+      print_info(const std::chrono::steady_clock::time_point &frame_timestamp, std::chrono::seconds interval_in_seconds) {
+        if (steady_clock::now() > data.last_print_time + interval_in_seconds) {
+          double latency_min_ms = data.latency_min / 10.;
+          double latency_max_ms = data.latency_max / 10.;
+          double latency_avg_ms = data.latency_total / 10. / data.calls;
+
+          auto fp = boost::format("%1$.1f");
+          BOOST_LOG(info) << "Latency (min/max/avg): " << fp % latency_min_ms << "ms/" << fp % latency_max_ms << "ms/" << fp % latency_avg_ms << "ms";
```

Comment:
If we're going to log this at the info level, we probably shouldn't do it so often, like maybe once a minute max.

I would also call it "Host Latency" or similar, like you did in the Moonlight Qt PR, just to avoid any possible confusion with end-to-end latency, network latency, etc.

Reply:
I planned to remove it before merge, needed to test it somehow because moonlight side is not merged. Probably should start marking such commits to avoid possible confusion.

Reply:
And on the topic of "host latency". Maybe we can come with something more fitting? If a frame gets delayed because of frame pacing sleep, that latency doesn't get included here. Maybe something like "host processing latency", or "host processing time". Parsec even calls "encoding".

Reply:
> I planned to remove it before merge, needed to test it somehow because moonlight side is not merged.

But maybe it makes sense to keep it after all, just move to Debug log level (it stays quiet during streaming), and maybe move the tracking class declaration. Will try to do it nicely.

Reply:
"Host Processing Latency" sounds fine to me (though technically, FEC and packetization is "processing" that isn't included in your calculation).

Reply:
> "Host Processing Latency" sounds fine to me

Alright, will rename it. We can extends packet header with "capture latency" later on. or hopefully get rid of that latency altogether in most configurations.

Reply:
Renamed "frame latency" to "frame processing latency". Will update moonlight pull requests right after we merge the sunshine part.

Moved stat tracking code to its own header so it can be reused. Frame processing latency is printed at debug log level every 20 seconds.

Reply:
> FEC encoding and packetization is "processing" that isn't included

I thought it was fast enough to neglect, and can't really inject stats into packet after encode? Alternatively we can delay it by one frame, and use the stats of the previous frame in packet header. But it would be getting quite convoluted.

Reply:
It mostly is, though we may have delays due to buffer exhaustion in the network stack if the outgoing link is slow. I don't think it's worth the effort to try to time those at this point.

Users should still see the indirect effects of the slow connection via the network latency and jitter stat even though we aren't directly measuring them here.